modules/pam_unix/support.c

```diff
@@ -641,11 +641,13 @@ _unix_blankpasswd (pam_handle_t *pamh, unsigned long long ctrl, const char *name
 }
 
 int _unix_verify_password(pam_handle_t * pamh, const char *name
-			  ,const char *p, unsigned long long ctrl)
+			  ,const char *pw, unsigned long long ctrl)
 {
 	struct passwd *pwd = NULL;
 	char *salt = NULL;
 	char *data_name;
+	char _p[MAX_PASS + 1] = "";
```

Comment:
Please use some different name not started with underscore. Also the initializer is redundant as you clear it with memset anyway.
You could also avoid renaming p to pw and just reuse the parameter.

Reply:
But `pw` it is a `const` so I cannot change it.

Reply:
It is a pointer to const data not a constant pointer. So you can freely change the pointer to point it at the copied truncated buffer.

Reply:
Thanks, I updated the PR accordingly. Looks better?

Reply:
Maybe also:
```diff --git a/modules/pam_unix/support.c b/modules/pam_unix/support.c
index 1171f63..e5415f5 100644
--- a/modules/pam_unix/support.c
+++ b/modules/pam_unix/support.c
@@ -787,6 +787,7 @@ int _unix_verify_password(pam_handle_t * pamh, const char *name
        }

 cleanup:
+       memset(pw, 0, sizeof(pw)); /* clear memory of the password */
        if (data_name)
                _pam_delete(data_name);
        if (salt)
```